src/xenia/cpu/compiler/passes/constant_propagation_pass.cc

```diff
@@ -253,12 +261,23 @@ bool ConstantPropagationPass::Run(HIRBuilder* builder) {
 
         case OPCODE_SELECT:
           if (i->src1.value->IsConstant()) {
-            if (i->src1.value->IsConstantTrue()) {
-              v->set_from(i->src2.value);
+            if (i->src1.value->type != VEC128_TYPE) {
+              if (i->src1.value->IsConstantTrue()) {
+                v->set_from(i->src2.value);
+                i->Remove();
+              } else if (i->src1.value->IsConstantFalse()) {
+                v->set_from(i->src3.value);
+                i->Remove();
+              } else if (i->src2.value->IsConstant() &&
+                         i->src3.value->IsConstant()) {
+                // TODO
+                // v->set_from(i->src2.value);
+                // v->Select(i->src3.value, i->src1.value);
+                // i->Remove();
+              }
             } else {
-              v->set_from(i->src3.value);
+              // TODO
```

Comment:
when you add TODO's you should put a description of what it is you need to do - like, 'TODO(DrChat): handle constant vec128 in src1' or something - that way, people scanning can find them and have a hope of implementing them easier